### DISCUSSION TITLE
Gradle plugin for RetroMCP

### DISCUSSION THREAD

**Discussion by 0xaa55h** [MCPHackers/RetroMCP-Java#91] [Ideas]
Hey whoever is reading this, I've got some experience writing plugins in Gradle(working but never really finished them, made a runtime plugin for Minecraft but it wasn't working due to its incapability to recompile), I'd like to create a plugin so you don't have to use the CLI or GUI jar files. I think this would be much easier to use/operate and edit. 
What do you think about it?

**Comment by 0xaa55h** [comment]
Also one more thing: Is there a Discord server for this? I feel like this is pretty bad way to communicate

**Reply by mudkipdev** [reply]
There is a RetroMCP channel in the Modification Station Discord. (You need to get the RetroMCP role first.)

https://discord.com/invite/8Qky5XY

**Comment by mudkipdev** [comment]
This was planned with the rewrite (separating the GUI and API layer) but that was abandoned in favor of adding things to the current codebase, so I'm guessing the idea is on hold for now

**Reply by 0xaa55h** [reply]
I've got no problem with doing it! Though it might take me some time, I'd absolutely fascinated to make something like this actually working, 
what I'd like to get is permission to do that

**Reply by mudkipdev** [reply]
I was planning on creating a separate project to decompile Minecraft using Gradle, only problem is that I have limited knowledge of making plugins despite using Gradle a lot, maybe if you're interested you can help

**Reply by 0xaa55h** [reply]
I might not have the best knowledge to work with Gradle API, I for sure can make something

**Reply by mudkipdev** [reply]
Join the Discord I linked
